### PR TITLE
Assembly Name Change for System.Transactions.Local

### PR DESCRIPTION
**Pull request by qizhanMS**
Rename assembly name to System.Transactions.Local.dll

### CODE REVIEW

**Comment by danmosemsft** [comment]
@weshaggard

File: src/System.Transactions.Local/ref/System.Transactions.Local.csproj [hunk 1]
```diff
@@ -9,6 +9,9 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp-Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Release|AnyCPU'" />
+  <PropertyGroup>
+    <AssemblyName>System.Transactions.Local</AssemblyName>
```
Comment (weshaggard) [inline src/System.Transactions.Local/ref/System.Transactions.Local.csproj]: Please remove this as you don't need to explicitly set the AssemblyName.

File: src/System.Transactions.Local/src/System.Transactions.Local.csproj [hunk 2]
```diff
@@ -11,6 +11,9 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcoreapp-Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Windows_NT-Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'uap-Windows_NT-Release|AnyCPU'" />
+  <PropertyGroup>
+    <AssemblyName>System.Transactions.Local</AssemblyName>
```
Comment (weshaggard) [inline src/System.Transactions.Local/src/System.Transactions.Local.csproj]: Not needed.

File: src/System.Data.Common/ref/System.Data.Common.csproj [hunk 3]
```diff
@@ -16,7 +16,7 @@
   <ItemGroup Condition="'$(TargetGroup)' == 'netfx'">
     <Reference Include="mscorlib" />
     <Reference Include="System" />
-    <Reference Include="System.Transactions" />
+    <Reference Include="System.Transactions.Local" />
```
Comment (weshaggard) [inline src/System.Data.Common/ref/System.Data.Common.csproj]: This still needs to remain System.Transactions as it is part of the netfx build configuration. That is what is breaking the current CI leg.

**Comment by karelz** [comment]
@qizhanMS can you please update title? It will populate the commit message.

**Comment by weshaggard** [comment]
>D:\j\workspace\AllConfigurat---c1564d53\Tools\Packaging.targets(869,5): error : Unable to find description for package System.Transactions.Local [D:\j\workspace\AllConfigurat---c1564d53\src\System.Transactions.Local\pkg\System.Transactions.Local.pkgproj]

Need to also update https://github.com/dotnet/corefx/blob/master/pkg/descriptions.json#L1878.

**Comment by qizhanMS** [comment]
@dotnet-bot test ci please

**Comment by qizhanMS** [comment]
@dotnet-bot rebuild please

**Comment by weshaggard** [comment]
>: error : PackageIndex from D:\j\workspace\AllConfigurat---c1564d53\pkg\Microsoft.Private.PackageBaseline\packageIndex.json is missing an entry for package System.Transactions.Local.  Please run /t:UpdatePackageIndex on this project to commit an update. [D:\j\workspace\AllConfigurat---c1564d53\src\System.Transactions.Local\pkg\System.Transactions.Local.pkgproj]

Looks like you also need to update the package index.

**Comment by qizhanMS** [comment]
@weshaggard How to update the package index?

**Comment by weshaggard** [comment]
From the error:

>. Please run /t:UpdatePackageIndex on this project to commit an update. 

goto \src\System.Transactions.Local\pkg\System.Transactions.Local.pkgproj and run `msbuild /t:UpdatePackageIndex`

We probably also want to manually remove the System.Transactions references in the index so perhaps it is better to manually change https://github.com/dotnet/corefx/blob/master/pkg/Microsoft.Private.PackageBaseline/packageIndex.json#L4658 to System.Transactions.Local.

**Comment by qizhanMS** [comment]
@weshaggard I got error when trying to run msbuild /t:UpdatePackageIndex on System.Transactions.Local.pkgproj.
On my machine, after build, I only have two nupkg: Microsoft.Private.CoreFx.NETCoreApp.4.4.0-beta-25024-0.nupkg and runtime.win7-x64.Microsoft.Private.CoreFx.NETCoreApp.4.4.0-beta-25024-0.nupkg.  And got file not found error when I run msbuild /t:UpdatePackageIndex. How can I get System.Transactions.Local.4.4.0-beta-25024-0.nupkg?

**Comment by weshaggard** [comment]
@qizhanMS 

The build configuration that does all the individual packages is the "allConfigurations" build leg. Can you run `build.cmd -allConfigurations` locally and then you should see the failure we are seeing in CI and you will also be able to build the individual pkgproj afterwards.

**Comment by qizhanMS** [comment]
@weshaggard I figure out the reference issue and update System.Data.Common.csproj and updatepackageindex for System.Transactions.Local. The build error this time is caused by a test failure. I don't think it is related to my change. There were xunit test failures in System.Diagnostics.Process.Tests. What's your suggestion for next step?

**Comment by weshaggard** [comment]
Innerloop Windows_NT Debug Build and Test

**Comment by weshaggard** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test please

**Pull request merged**